scx_loader: Lockdown systemd service

frap129 opened the pull request:
This is an attempt to perform the systemd hardening suggested by #12 . The following options were determined by profiling the service with [shh (systemd hardening helper)](https://github.com/desbma/shh). It was done by starting profiling with shh, running through every scxctl option (stop, start sched with modes/args, switch with modes/args, restart, get, list), then ending profiling. AFAICT everything works fine w/ the new hardening options, but I would appreciate testing to make sure it works on more than just my system.

frap129 committed:
scx_loader: Lockdown systemd service

sirlucjan commented:
```RestrictAddressFamilies=AF_UNIX
SocketBindDeny=ipv4:tcp
SocketBindDeny=ipv4:udp
SocketBindDeny=ipv6:tcp
SocketBindDeny=ipv6:udp
```

I'm not sure about above, but LGTM overall.

vnepogodin reviewed (changes requested):
^

vnepogodin committed:
services/scx_loader: fix issue with rustland

scx_loader[2715]: [INFO]: starting scx_rustland command
scx_loader[70589]: thread 'main' (70589) panicked at rust/scx_rustland_core/src/alloc.rs:495:17:
scx_loader[70589]: mlockall failed with error code: -1
scx_loader[70589]: note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
scx_loader[2715]: [ERROR]: Failed to start scheduler (attempt 5/5)

Headcrabed commented:
@vnepogodin I'm just using scx_loader with systemd service to do auto-loading. Kernel is Fedora's official pre-built 6.18.10, and scx_loader comes from `scx-tools-git` in https://copr.fedorainfracloud.org/coprs/bieszczaders/kernel-cachyos-addons/

scx_loader config:
```
# This field specifies the scheduler that will be started automatically when scx_loader starts (e.g., on boot).
default_sched = "scx_lavd"

# This field specifies the mode which will be used when scx_loader starts (e.g., on boot).
default_mode = "Gaming"

# This "structure" allows configuring flags for each scheduler mode of particular scx scheduler
[scheds.scx_lavd]
auto_mode = []
gaming_mode = ["--performance"]
lowlatency_mode = ["--performance"]
powersave_mode = ["--powersave"]
server_mode = []
```

The scheduler I'm using needs root to run, but with this PR seems it's not running as root. Reverting this commit or adding `User=root` to service file both works.

Headcrabed commented:
@vnepogodin any updates？
